### PR TITLE
[Llama3] Send decode output logits to dram to reduce trace l1 usage and fix 8b-n150 memory crash

### PR DESCRIPTION
**Pull request by skhorasganiTT**
### Ticket
https://github.com/tenstorrent/tt-metal/issues/16692

### Problem description
- Llama8B-N150 would crash with an out of l1 error when executing prefill with prompt lengths >= 2k after completing a decode trace
- This was due to the trace capture reserving a large amount of l1 space for the output logits, causing subsequent prefills (specifically, the first op in attention, the qkv matmul) to run out of l1

### What's changed
- Moved the decode output logits to DRAM so that subsequent operations could use the l1 space
- Note: this results in a ~1% drop in e2e perf for the n150/n300 llama3 text models

### Checklist
- [x] Post commit CI passes
- [x] Blackhole Post commit (if applicable)
- [x] Model regression CI testing passes (if applicable)
- [x] Device performance regression CI testing passes (if applicable)
- [x] **(For models and ops writers)** Full [new models](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml) tests passes
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

**Comment by skhorasganiTT** [comment]
T3K demo tests: https://github.com/tenstorrent/tt-metal/actions/runs/12876364064

**Review by cglagovichTT — APPROVED** [review]
Since this has a perf hit, do you want to apply this change to all Llama models or only the system x Llama-variants which are L1 OOM in trace?

**Comment by skhorasganiTT** [comment]
> Since this has a perf hit, do you want to apply this change to all Llama models or only the system x Llama-variants which are L1 OOM in trace?

Since the perf hit is quite small I initially elected to make it consistent across the models but I'm also okay to make it conditional on 8b-n150. @mtairum @yieldthought do you have a preference on this?

**Comment by mtairum** [comment]
Honestly, for 1% drop in N150/N300, I won't fuss much, given that it brings back stability. 

Perf doesn't mean much if things don't work 😛

**Review by mtairum — APPROVED** [review]
